### ISSUE TITLE
Referencing images creates invalid ePub v2

### ISSUE THREAD

**Issue by kikofernandez** [lierdakil/pandoc-crossref#200]
The reference of an image, e.g. `![](MEDIA/assets/art-artist-artistic-316465.jpg){#figure:intro}`, when converted to ePub v2, creates the following HTML:

```html
<p><img src="../media/file0.jpg" id="figure:intro" /></p>
```

When I run the epub checker `epubcheck-4.0.2`, it throws the following exception:

```
Validating using EPUB version 2.0.1 rules.
WARNING(CSS-006): book.epub/EPUB/styles/stylesheet1.css(204,4): CSS position:fixed property should not be used in EPUBs.
ERROR(RSC-005): book.epub/EPUB/text/ch003.xhtml(21,54): Error while parsing file 'value of attribute "id" is invalid; must be an XML name without colons'.
ERROR(RSC-005): book.epub/EPUB/text/ch003.xhtml(21,54): Error while parsing file 'element "img" missing required attribute "alt"'.
ERROR(RSC-005): book.epub/EPUB/text/ch003.xhtml(39,43): Error while parsing file 'value of attribute "id" is invalid; must be an XML name without colons'.

Check finished with errors

epubcheck completed
```

It seems to complain that the attribute `id` is not valid and that the `alt` property is missing.

**Comment by K4zuki** [comment]
@kikofernandez Try `#fig:intro` instead of `#figure:intro`

Reference: <http://lierdakil.github.io/pandoc-crossref/#image-labels>

**Comment by kikofernandez** [comment]
the error is exactly the same, but thanks

**Comment by lierdakil** [comment]
Frankly, it's your fault that `alt` property is missing. You didn't specify any, and software can't generate a meaningful `alt` text from images without user input (at least not reliably and not on a desktop computer anyway).
As for the other one, in my experience, it doesn't really matter. It is theoretically possible to replace colons with other characters, but it's an open question whether this should be done in pandoc-crossref or pandoc itself.
In any case, this isn't a pandoc-crossref specific issue, although I will admit that pandoc-crossref does kinda force you to use colons.
TL;DR: mostly safe to ignore, fix viability is unclear, can't give an ETA. The best I can offer is that I'll keep this in mind when working on the next major release (why major? because any kind of "fix" I can think of from the top of my head would imply a major change in behaviour)

**Comment by hhirtz** [comment]
If you don't mind the bump, i would like to share a workaround. The issue in my case is that colons do not work with cleveref with french latex output, for some reason[^1] (pandoc-crossref v0.3.16.0 & pandoc v3.1.2).

Anyway, this lua filter replaces the colons from pandoc-crossref labels with dashes and makes it work for latex output. For HTML you would need to change the regexs in `function replaceRef` i think.

```lua
-- crossrefdash.lua

function CodeBlock(elem)
    elem.identifier = elem.identifier:gsub("^lst:", "lst-")
    return elem
end
function Figure(elem)
    elem.identifier = elem.identifier:gsub("^fig:", "fig-")
    return elem
end
function Header(elem)
    elem.identifier = elem.identifier:gsub("^sec:", "sec-")
    return elem
end

function Blocks(elem)
    local function isCrossref(t)
        return t == "fig" or t == "eq" or t == "tbl" or t == "sec" or t == "lst"
    end

    local function replaceRef(cmd, s)
        return s:gsub("\\"..cmd.."{(.*):(.*)}", function(reftype, refname)
            if isCrossref(reftype) then
                return "\\"..cmd.."{"..reftype.."-"..refname.."}"
            end
        end)
    end

    return elem:walk({
        RawInline = function (elem)
            elem.text = replaceRef("labelcref", elem.text)
            elem.text = replaceRef("label", elem.text)
            elem.text = replaceRef("cref", elem.text)
            elem.text = replaceRef("Cref", elem.text)
            return elem
        end
    })
end
```

Use with `--lua-filter crossrefdash.lua`. This option must appear after `--filter pandoc-crossref`.

[^1]: > "Cleveref can’t cope with active characters being present in cross-reference label names. For example, if French babel support is loaded, the commonly used “:” in label names will often fail, spewing the usual random selection of mysterious TEX errors that accompany such deep-seated errors. The solution is to avoid using active characters in label names. (You may need to consult the babel documentation to discover which active characters are defined in your language.)"
http://tug.ctan.org/tex-archive/macros/latex/contrib/cleveref/cleveref.pdf
